### ISSUE TITLE
Simple control experiment rgb_depth_system does not give expected results

### ISSUE THREAD

**Issue by miguelriemoliveira** [lardemua/atom#672]
Tagging @danifpdra , @manuelgitgomes , @Kazadhum for visibility.

Working on #642 we get a final depth to rgb evaluation of 2 pixels of error. This does not seem bad, but since this is a simulated system I wounder if we should be having more. Also, @danifpdra reports getting evaluations of this type around 1 pixel, so I am creating this issue to investigate.

The experiment is very simple.

   - 1 collection only
   -  using as train the **train dataset dataset_corrected.json**, which contains initial transformation values, which are perfect since this is a simulation 
   -  carry out evaluation using:
       - as train use the **train dataset dataset_corrected.json**
       - as test the same **train dataset dataset_corrected.json**

This means that we are "pseudo" calibrating for a single collection, and have the perfect ground truth transformation. 
Then we evaluate on a test dataset which is actually the same as used for training.

The error in this case should be close to zero, right? Am I making any mistake?

**How to run:**

run this:

    clear && rosrun atom_evaluation depth_to_rgb_evaluation -train_json $ATOM_DATASETS/rgb_depth_system/rgb_depth_system_example_train_dataset/dataset_corrected.json --test_json $ATOM_DATASETS/rgb_depth_system/rgb_depth_system_example_train_dataset/dataset_corrected.json -ds depth -cs rgb -csf "lambda x: int(x) in [0]" -si

**Comment by miguelriemoliveira** [comment]
The results are:

```terminal
Errors per collection
+--------------+-----------+-------------+-------------+
| Collection # | RMS (pix) | X err (pix) | Y err (pix) |
+--------------+-----------+-------------+-------------+
|     000      |   4.5850  |    2.1880   |    2.2055   |
|   Averages   |   4.5850  |    2.1880   |    2.2055   |
+--------------+-----------+-------------+-------------+
```

this is very far from a perfect result. The image shows:

![image](https://github.com/lardemua/atom/assets/11350998/739adfa0-ce59-41a5-a8b5-47d57fcd1d48)

it seems the error is only on the bottom ...

![image](https://github.com/lardemua/atom/assets/11350998/1f683e1e-73cd-4568-85ea-2469f497a285)

**Comment by miguelriemoliveira** [comment]
The labels for the depth image in fact seem to be wrong in the bottom

![image](https://github.com/lardemua/atom/assets/11350998/7457e279-9c40-4310-9a40-85d68d43a7ce)

**Comment by miguelriemoliveira** [comment]
I improved the depth boundary labeling 

![image](https://github.com/lardemua/atom/assets/11350998/bc2702de-5444-4b59-a856-6127f8f11beb)

and rerun the evaluation. I get this:

![image](https://github.com/lardemua/atom/assets/11350998/d01b0a6d-80f0-4853-8268-bc61cc90bc9b)

which is much better.

**Comment by miguelriemoliveira** [comment]
With a better visualization we have:

![image](https://github.com/lardemua/atom/assets/11350998/4598283f-6952-40ca-87cb-9eba1e358599)

**Comment by miguelriemoliveira** [comment]
I think there might be a bug here ... doing a preposterous annotation

![image](https://github.com/lardemua/atom/assets/11350998/7df29d60-63d3-481b-9a1a-f046d9ccb9db)

gets this

![image](https://github.com/lardemua/atom/assets/11350998/91148068-1de4-46a2-bd2c-936cb719557e)


I am not sure this is correct. Will think about it later ...

**Comment by danifpdra** [comment]
Hi @miguelriemoliveira,

Why do you think that's not correct? That seems correct to me

**Comment by miguelriemoliveira** [comment]
Here's a better visualization. You are right, its ok.

Green -> the annotated points
red -> the projected points
yellow -> the line found from each projected point to the closest annotated point

![image](https://github.com/lardemua/atom/assets/11350998/9b4aff5e-5911-4aba-bb93-ab1cc4bbb772)

**Comment by miguelriemoliveira** [comment]
Yeah, it looks good. So if I annotate the bottom line a bit to the inside of the pattern

![image](https://github.com/lardemua/atom/assets/11350998/3c61550f-9519-4cc9-b818-aa2f51380923)


we get this:

![image](https://github.com/lardemua/atom/assets/11350998/c1884952-2ea3-426d-824e-7148aecbeffd)

which results in table:

![image](https://github.com/lardemua/atom/assets/11350998/9d0af3b9-0abe-4605-abb8-f25f48f26f45)


I think I am convinced that I am sure that I am confident that these errors are a natural part of the process, and not because of some bug. 

@danifpdra , are you also convinced that you are sure that you are confident?

Should I stop investigating this? I think I should ...

**Comment by danifpdra** [comment]
> Yeah, it looks good. So if I annotate the bottom line a bit to the inside of the pattern
> 
> ![image](https://user-images.githubusercontent.com/11350998/278864324-3c61550f-9519-4cc9-b818-aa2f51380923.png)
> 
> we get this:
> 
> ![image](https://user-images.githubusercontent.com/11350998/278864351-c1884952-2ea3-426d-824e-7148aecbeffd.png)
> 
> which results in table:
> 
> ![image](https://user-images.githubusercontent.com/11350998/278864371-9d0af3b9-0abe-4605-abb8-f25f48f26f45.png)
> 
> I think I am convinced that I am sure that I am confident that these errors are a natural part of the process, and not because of some bug.
> 
> @danifpdra , are you also convinced that you are sure that you are confident?
> 
> Should I stop investigating this? I think I should ...

I am absolutely convinced that I'm sure that I'm confident, yes.

I reran all my experiments from camerabot (RGB-D) real and simulated and here's my results. It all seems good to me 

![image](https://github.com/lardemua/atom/assets/47574905/db35d806-2e2e-435c-92eb-fbc92429e295)

**Comment by miguelriemoliveira** [comment]
OK, thanks @danifpdra . Closing the issue.
